src/common/helpers/data-extract/data-extract.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @typedef  { 'name' | 'file' | 'text' |' radio' | 'address' | 'checkbox' } AnswerType
+ * @typedef  {{ type: AnswerType, value: string, displayText: string }} AnswerData
+ * @typedef  {{ question: string, questionKey: string, answer: AnswerData }} QuestionAnswerData
+ * @typedef  {{ title: string, sectionKey: string, questionAnswers: QuestionAnswerData[]}} SectionData
+ */
+
+/**
+ * @param {string} questionKey
+ * @param {string} sectionKey
+ * @param {SectionData[]} sections
+ * @returns {QuestionAnswerData|undefined}
+ */
+export const getQuestionFromSections = (questionKey, sectionKey, sections) => {
+  const section = Array.isArray(sections)
+    ? sections?.find((section) => section.sectionKey === sectionKey)
+    : {}
+  const question = Array.isArray(section?.questionAnswers)
+    ? section?.questionAnswers?.find(
+        (question) => question.questionKey === questionKey
+      )
+    : undefined
+  return question
+}
+
+/**
+ * @param {object} payload
+ * @returns {SectionData[]|undefined}
+ */
+export const getSectionsFromPayload = (payload) => {
```

Comment:
Same with this - we could specify the type as just `SectionData[]`.  If you're worried about malformed data getting in, then I'd argue it's time for Joi assertions, so we know data is well structured at the boundary & don't have to be too defensive.

Reply:
We could do this as part of the ticket that will do the schema validation?

Reply:
yup - I'd suggest "assume the data is good for now", then we can make sure it's good in the Joi validation ticket. Let's us keep simple code all the way though.

Does that work for you?

Reply:
done